### ISSUE TITLE
[Mining] Statistics

### ISSUE THREAD

**Issue by NovaT82** [tari-project/tari-launchpad#22]
![image](https://user-images.githubusercontent.com/9142942/167814607-4d3fd0a8-50bd-4161-98a2-e51defc38375.png)

## ac:
- tari statistics 
- monero mining statistics are not going to be available, it's too complicated to get this on initial launch
- data should be equal to sum of `amount`s for WalletTransaction events with `event` field with value `'mined'`

## filters:
_Monthly_ filter: as on the screenshot above, if we only have data for one month, don't show "vs" comparisons
_Yearly_ filter:  should show all months in the selected year, with values of mined xtr, xmr for each month, if we only have data for one year, don't show "vs" comparisons
_All_ filter:  should be **disabled** if we only have data for only one year, otherwise show year by year results, don't show "vs" indicator on coin balances (nothing to compare)

**Comment by NovaT82** [comment]
#### tarnas14 commented:

@alex1307 @CjS77 

I assume blockchain stores the information about mined coins by each node
should we assume that this information is available from backend and we can just request it?

**Comment by NovaT82** [comment]
#### tarnas14 commented:

question ~pending~ about "all" and "yearly" filters
*answer included in description*
![image](https://user-images.githubusercontent.com/9142942/167814607-4d3fd0a8-50bd-4161-98a2-e51defc38375.png)

**Comment by NovaT82** [comment]
#### CjS77 commented:

There are 2 ways of getting "blocks mined" information:
- By querying UTXOs that are both coinbase UTXOs and are mined / confirmed. (This gets you history)
- By signing up for notifications. We currently only allow notifications via a shell script, but are looking into piping the events directly to the Tauri FE (This gets you live events)

**Comment by NovaT82** [comment]
#### CjS77 commented:

@alex1307 - We should add Tauri commands that make this a simple call. The FE should not be wrangling this data.
Bear in mind that the historical events call is somewhat expensive, so some discussion on the API would serve us well.

How often will you be requesting the full history?
Can we assume the FE caches previous calls, so we can return new changes since a given timestamp?

**Comment by NovaT82** [comment]
#### tarnas14 commented:

since I don't think historical data ever changes in blockchain, it wouldn't hurt us to save the data somewhere

is it possible to make an API that would return results with daily resolution, called for specific windows, e.g. `from=2022-04-01&to=2022-04-30`?
FE could then save the results internally and make sure never to call for that window again (assuming the window is in the past)
and then when the user changes filters on the statistics view, and we don't have data to show to them, we would call the api for missing windows?

I don't know how the UTXO is queried and how expensive this API is, so all suggestions are welcome.

**Comment by NovaT82** [comment]
#### tarnas14 commented:

@CjS77 @alex1307 do you have any new info on this?

**Comment by NovaT82** [comment]
#### CjS77 commented:

@alex1307 is still busy with this.

**Comment by NovaT82** [comment]
#### tarnas14 commented:

ok, I'll put backend integration on hold and wait till we get any info on the interface (y)

**Comment by NovaT82** [comment]
#### tarnas14 commented:

~backend contract: GET method `from=<Date>&to=<Date>&interval='monthly'|'yearly'|'all'`~

~interval is optional, if not provided, data will be returned with values per day~
~if it is provided, it will combine data `yearly` -> values per month, `all` -> values per year~

UPDATE:
frontend will listen on transaction events and store them in sqlite (after #46 implements it)
